src/terminal/terminal.rs

```diff
@@ -234,39 +234,142 @@ where
         Ok(())
     }
 
-    /// Synchronizes terminal size, calls the rendering closure, flushes the current internal state
-    /// and prepares for the next draw call.
+    /// Draws a single frame to the terminal.
     ///
-    /// This is the main entry point for drawing to the terminal.
+    /// Returns a [`CompletedFrame`] if successful, otherwise a [`std::io::Error`].
```

Comment:
```suggestion
    /// Returns a Ok([`CompletedFrame`]) if successful, otherwise a [`std::io::Error`].
```

Reply:
```rust
    /// Returns [`Result::Ok`] containing a [`CompletedFrame`] if successful, otherwise
    /// [`Result::Err`] containing the [`std::io::Error`] that caused the failure.
```

Reply:
Result is a basic concept of Rust and this seems a bit too much explanation of that. I would just go with CompletedFrame or Error. Maybe "Returns a Result containing either …".

Reply:
I checked the std lib and it varies in the explanation. This is one of the more verbose versions copied and paraphrased from one of the std lib methods (In string / vec / result).